tasks/retrieval_deal.go

```diff
@@ -85,15 +89,32 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
 	}
 
 	dealStage = AddLog(dealStage, "deal sent to miner")
-	err = updateStage("ProposeDeal", dealStage)
+	err = updateStage(stage, dealStage)
 	if err != nil {
 		return err
 	}
 
 	lastStatus := retrievalmarket.DealStatusNew
-	var lastBytesReceived uint64 = 0
+	var lastBytesReceived uint64
+	var prevStage string
+	var stageTimeout <-chan time.Time
```

Comment:
the chan can be declared inside the loop, no? you can then avoid the `stageTimeout = nil` assignment too, as it's already the zero value.

Reply:
this would also make the code easier to follow - the chan var being shared by different iterations initially made me think that the channel would be reused, when I'm pretty sure it's not

Reply:
Done